### ISSUE TITLE
Topics list fail to load if started from an offline state

### ISSUE THREAD

**Issue by codebu5ter** [zulip/zulip-mobile#2310]
**Steps to reproduce:**

1. Go offline.
2. Narrow to any stream.
3. Tap the 'bullet list' icon to go to the `Topics` screen.
4. Turn on mobile data/Wi-Fi.
5. The list of Topics fail to load even after turning on mobile data.

![screenshot_20180413-223623](https://user-images.githubusercontent.com/16636569/38748607-4d3d46da-3f6c-11e8-8947-20abd0f7b31b.png)

**Comment by pulkonet** [comment]
I think this is related to #2287 .

**Comment by codebu5ter** [comment]
@pulkonet That issue is similar, but since this deals with the display of topics while that issue deals with the display of messages, I think both need separate issues.

**Comment by pulkonet** [comment]
You're right, but I stated the relation just so that anyone trying to fix it could take reference.

**Comment by roberthoenig** [comment]
Hmm, this issue seems to involve two bugs: One bug is the RN `fetch() `bug that causes #2287. But even if that bug was resolved, I think this issue would still be valid, since it is about narrowing to a topic *while* the device is offline and coming back online afterwards. Right now, I don't think we resend requests that have been made while offline. In any case, we probably have to wait until https://github.com/facebook/react-native/issues/19709 gets resolved.
